ADR/0027-availability-probe-framework.md

```diff
@@ -0,0 +1,125 @@
+# 27. Availability Probe Framework
+
+* Date 2023-07-06
+
+## Status
+
+Accepted
+
+## Context
+
+As an AppStudio developer building functionality for the platform, I want to be able to
+easily visualize and comprehend the stability and availability of deployed systems in
+order to inform and influence future work towards improving the overall system
+reliability.
+  
+Such indication should tell us the overall up time of AppStudio with respect to services
+under the control of AppStudio developers.
+
+AppStudio is defined to be available at a given moment if all of its components are
+reporting to be available at that given moment.
```

Comment:
Are we only considering available/unavailable or is there also an intermediate degraded state? What does it mean for AppStudio to not be available?

Reply:
> Are we only considering available/unavailable or is there also an intermediate degraded state?

We're only considering available/unavailable, but there will be visibility into the components causing it to be unavailable.
This is per the feature refinement process: https://docs.google.com/document/d/1ar8PJX4MWmw7fpEgB4cxK7d8lXOpkojdTKrJPsok1A4/edit?usp=sharing

> What does it mean for AppStudio to not be available?

it means that there's at least one component which reports to be unavailable. In reality, it could mean any level of (un)availability - anything between dead to almost fully working.

Reply:
This is likely counterintuitive for those that are familiar with availability metrics, so we should ensure that we clarify this point. I added degraded to out of scope in the document.